### PR TITLE
CLDR-14311 use JDK11 for build [v42]

### PR DESCRIPTION
**Pull request by srl295**
CLDR-14311

Bump to JDK 11.  To be landed in CLDR v42

### CODE REVIEW

**Comment by srl295** [comment]
@btangmu a JDK11 build passed, may be J9 though.

**Comment by btangmu** [comment]
FYI, here's some of what I did to install  adoptopenjdk-openj9 on macOS with homebrew (homebrew must already be installed):

```
brew install jenv
brew install adoptopenjdk-openj9
jenv enable-plugin export
ls -la ~/.jenv/versions
jenv global 15
jenv versions
```

Also I added this to my `.bash_profile`:
```
# jenv -- for switching java versions with brew
export PATH="$HOME/.jenv/bin:$PATH"
eval "$(jenv init -)"
```

**Comment by srl295** [comment]
probably land this in v42

File: .github/workflows/build-icu.yml [hunk 1]
```diff
@@ -32,10 +32,10 @@ jobs:
           ref: ${{ github.event.inputs.cldr-ref }}
           lfs: false
           repository: ${{ github.event.inputs.cldr-repo }}
-      - name: Set up JDK 1.8
+      - name: Set up JDK 11
```
Comment (btangmu) [inline .github/workflows/build-icu.yml:35]: does the name need the version number, or would "Set up JDK" do as well?

**Review by btangmu — DISMISSED** [review]

File: .github/workflows/build-icu.yml (hunk 1 above)
Comment (srl295) [reply .github/workflows/build-icu.yml:35]: done

**Comment by pedberg-icu** [comment]
The ticket still says v41, should that be updated?

**Review by pedberg-icu — APPROVED** [review]

**Review by btangmu — APPROVED** [review]

**Pull request merged**